### ISSUE TITLE
Model has higher validation error when calling clearState() before

### ISSUE THREAD

**Issue by marcel1991** [torch/nn#1179]
I found that every time when I save my model the validation loss gets higher than it was on previous steps without saving the model. After investigating this strange behaviour I found that the problem is due to calling clearState() (when saving the model) before running the validation procedure. When running the validation I call model:evaluate(). Then I return back with model:training() to the training.

I am now wondering how I can go back properly to the training after I called clearState() since the validation loss indicates that after calling clearState() once the model somehow is not in the same state as before anymore.
